admin/server/auth/claims.go

```diff
@@ -247,3 +190,112 @@ func unionProjectRoles(a *adminv1.ProjectPermissions, b *database.ProjectRole) *
 		ManageProjectMembers: a.ManageProjectMembers || b.ManageProjectMembers,
 	}
 }
+
+// organizationPermissionsUser is a Claims implementation for resolve the organization permissions for user
+func (c *authTokenClaims) organizationPermissionsUser(ctx context.Context, orgID string) *adminv1.OrganizationPermissions {
+	c.Lock()
+	defer c.Unlock()
+
+	if perm, ok := c.orgPermissionsCache[orgID]; ok {
+		return perm
+	}
+
+	roles, err := c.admin.DB.ResolveOrganizationRolesForUser(context.Background(), c.token.OwnerID(), orgID)
+	if err != nil {
+		panic(fmt.Errorf("failed to get org permissions: %w", err))
+	}
+
+	composite := &adminv1.OrganizationPermissions{}
+	for _, role := range roles {
+		composite = unionOrgRoles(composite, role)
+	}
+
+	c.orgPermissionsCache[orgID] = composite
+	return composite
+}
+
+// organizationPermissionsService is a Claims implementation for resolve the organization permissions for service
+func (c *authTokenClaims) organizationPermissionsService(ctx context.Context, orgID string) *adminv1.OrganizationPermissions {
+	// Not adding caching for now
+	var orgPermissions *adminv1.OrganizationPermissions
+	service, err := c.admin.DB.FindService(ctx, c.token.OwnerID())
+	if err != nil {
+		panic(fmt.Errorf("failed to get service info: %w", err))
+	}
+
+	if orgID == service.OrgID {
+		orgPermissions = &adminv1.OrganizationPermissions{
+			ReadOrg:          true,
+			ManageOrg:        true,
+			ReadProjects:     true,
+			CreateProjects:   true,
+			ManageProjects:   true,
+			ReadOrgMembers:   true,
+			ManageOrgMembers: true,
+		}
+	}
+	return orgPermissions
```

Comment:
Do a `return` on line 227 and then on line 237 do `return &adminv1.OrganizationPermissions{}` so that the return value is never `nil`